src/Database/Retry/ReconnectStrategy.php

```diff
@@ -110,7 +110,9 @@ protected function reconnect(): bool
 
         try {
             $this->connection->connect();
-            $this->connection->log('[RECONNECT]');
+            if ($this->connection->isQueryLoggingEnabled()) {
+                $this->connection->log('[RECONNECT]');
```

Comment:
We should probably add a wrapper to log if a logger is attached.

Reply:
"_to_" as in _inside_ of `log()`, or as in _around_` (ie an additional method)?

Reply:
Don't we want it to scream at people if no logger is available when something tries to log something?

Reply:
I would say no. If you want logging, attach a logger.

Reply:
All I'm saying is put the is logging enabled check inside the logging function. We should still error if creating the logger failed.

Reply:
I see, I thought you meant to prevent the latter. I guess it could make sense to have the method obey the configuration, but since it's a public method, we probably can't change its behavior right now?

Reply:
It's not blocking this PR, no.